jquery.flot.navigate.js

```diff
@@ -384,27 +386,56 @@ can set the default in the options.
 
             return delta;
         }
+
+        var isDiagonalMode = function(delta) {
+            if (Math.abs(delta.x) > 0 && Math.abs(delta.y) > 0) {
+                return true;
+            }
+            return false;
+        }
+
+        var restoreAxisOffset = function(axes, initialState, delta) {
+            var axis;
+            Object.keys(axes).forEach(function(axisName) {
+                axis = axes[axisName];
+                if (delta[axis.direction] === 0) {
+                    axis.options.offset.below = initialState[axisName].navigationOffset.below;
+                    axis.options.offset.above = initialState[axisName].navigationOffset.above;
+                }
+            });
+        }
+
         var prevDelta = { x: 0, y: 0 };
         plot.smartPan = function(delta, initialState, panAxes, preventEvent) {
-            var snap = shouldSnap(delta);
+            var snap = shouldSnap(delta),
+                axes = plot.getAxes();
             delta = adjustDeltaToSnap(delta);
 
+            if (isDiagonalMode(delta)) {
+                initialState.diagMode = true;
```

Comment:
Maybe on line only:
initialState.diagMode = isDiagonalMode(delta);

Reply:
If we do this, when we return from diagonal mode and just snap,  this variable will become false and we will not restore the other axis offset